microsetta_private_api/repo/source_repo.py

```diff
@@ -307,3 +312,56 @@ def check_source_post_overhaul(self, account_id, source_id):
                 (account_id, source_id)
             )
             return cur.rowcount == 1
+
+    def get_external_reports(self, source_id):
+        """Retrieve list of external reports for a given source, if any
+        NB: We only extract certain columns because there's no need to waste
+        resources on the actual file_contents until the user hits the path
+        to view/download the report
```

Comment:
```suggestion
        """Retrieve list of external reports for a given source, if any
        
        We only extract certain columns because there's no need to waste
        resources on the actual file_contents until the user hits the path
        to view/download the report
```

The cost of pulling this from the database is really small though, and introduces multiple paths to pull rows from this table, unless the plan is to store many MB or GB of data per row (which I do not believe is the case, and not what the design here is oriented toward). If the data isn't needed for the client, why not just have the API method delete it?